### PR TITLE
Use correct image to calculate tracks in sample space

### LINKED ISSUE #38: Strange results when tracing in Sample Space
Hi!
I've used your tool to trace fluorescent electrode tracks. As this works perfectly in standard space, somehow it does not properly work in sample space. 
Here you can see traces that start in V1 and end in Zona Incerta. When I do this in Sample space the fits tell me the track is starting in area A, and ending in Globus Pallidus Externa, regions more anterior from what is actually true.

I noticed the 3D view of sample space already shows a slightly wrong overlay of the atlas then when you look at the slices in 2D (see images)

If I do the same in standard space (see other images), I get proper fits, and also the output files (csv) tell me the correct area names.

Thanks for your time!

3D_SampleSpace
Example_SampleSpace
Example_StandardSpace
Example_StandardSpace2

### PR DESCRIPTION
**Pull request by adamltyson**
Fixes #38

### CODE REVIEW

File: brainreg_segment/tracks/analysis.py
Comment (dstansby) [inline brainreg_segment/tracks/analysis.py:1]: Are any of the updated functions ones that users are likely to use in isolation? If it is, it's worth starting a (or adding to an existing) changelog and noting the breaking change to the function signature.

File: brainreg_segment/tracks/analysis.py [hunk 1]
```diff
@@ -91,27 +95,30 @@ def run_track_analysis(
...
     """
     spline_regions = []
-    for p in spline.tolist():
+    for coord in spline.tolist():
         try:
-            spline_regions.append(
-                atlas.structures[atlas.structure_from_coords(p)]
-            )
+            coord = [int(c) for c in coord]
+            atlas_value = atlas_layer_data[tuple(coord)]
```
Comment (dstansby) [inline brainreg_segment/tracks/analysis.py:120]:
```suggestion
            coord = (int(c) for c in coord)
            atlas_value = atlas_layer_data[coord]
```
Saves a few characters

**Review by dstansby — COMMENTED** [review]
Looks good overall, with a couple of suggestions. Is there a unit test of `analyse_track_anatomy` that can be added to make sure this doesn't break in the future?

File: brainreg_segment/tracks/analysis.py
Comment (adamltyson) [reply brainreg_segment/tracks/analysis.py:1]: I doubt anyone is going to use any of the `brainreg-segment` code directly. Changelog is a good idea though. How we do this is probably part of a larger conversation about whether the "product" is BrainGlobe, the individual tools, or some combination. The changelog should reflect this.

Comment (dstansby) [reply brainreg_segment/tracks/analysis.py:1]: If this is just backend code, then 👍 to just merging this without any changelog here. When we do the docs one other thing we should do is be clear about what is and isn't public facing API, so we and users know what we can change without warning.

File: brainreg_segment/tracks/analysis.py (hunk 1 above)
Comment (adamltyson) [reply brainreg_segment/tracks/analysis.py]: Doesn't work though :( You can't index with a generator.

**Comment by adamltyson** [comment]
> Looks good overall, with a couple of suggestions. Is there a unit test of `analyse_track_anatomy` that can be added to make sure this doesn't break in the future?

A unit test wouldn't have caught this unfortunately. The wrong data was passed into it from initialisation of the widget. I've added a specific test for this function anyway.

**Comment by adamltyson** [comment]
@dstansby the only test I can think of to prevent something like this is a full end-to-end test of the entire users workflow (in both atlas and sample space). Curious as to your thoughts, but I'm not sure adding this is helpful as it's likely to be very sensitive to any changes we make to any part of the code.

**Review by dstansby — APPROVED** [review]

**Pull request merged**